include/seqan3/range/view/pairwise_combine.hpp

```diff
@@ -0,0 +1,618 @@
+     * ### Exceptions
+     *
+     * No-throw guarantee.
+     */
+    constexpr iterator_type end() noexcept
+    {
+        return iterator_type{back_iterator, std::ranges::begin(src_range), std::ranges::end(src_range)};
+    }
+
+    //!\copydoc end()
+    constexpr iterator_type end() const noexcept
+    {
+        return iterator_type{back_iterator, std::ranges::begin(src_range), std::ranges::end(src_range)};
+    }
+
+    //!\copydoc end()
+    constexpr iterator_type cend() const noexcept
+    {
+        return end();
+    }
+    //!\}
+
+private:
+
+    //!\brief The underling range.
+    source_range_type src_range{};
+    //!\brief The cached iterator pointing to the last element of the source range.
+    std::ranges::iterator_t<source_range_type> back_iterator{};
+};
+
+/*!\name Type deduction guides
+ * \{
+ */
+
+//!\brief Deduces the correct template type from a view.
+template <std::ranges::View other_range_t>
+pairwise_combine_view(other_range_t range) -> pairwise_combine_view<other_range_t>;
+
+//!\brief Deduces the correct template type from a non-view lvalue range by wrapping the range in seqan3::view::all.
+template <std::ranges::ViewableRange other_range_t>
+pairwise_combine_view(other_range_t && range) ->
+    pairwise_combine_view<decltype(view::all(std::declval<other_range_t &&>()))>;
+//!\}
+
+} // namespace seqan3::detail
+
+namespace seqan3::view
+{
+/*!\name General purpose views
+ * \{
+ */
+
+/*!\brief             A view adapter that generates all pairwise combinations of the elements of the underlying range.
+ * \tparam urng_t     The type of the range being processed. See below for requirements.
+ * \param[in] urange  The range being processed.
+ * \returns           A view over all pairwise combinations of the elements of the underlying range.
+ *                    See below for the properties of the returned range.
+ * \ingroup view
+ *
+ * \details
+ *
+ * This view generates two-element tuples representing all unique combinations of the elements of the underlying range
+ * (the order of the elements does not matter). If the underlying range has less than two elements the returned range is
+ * empty, otherwise the size of the returned range corresponds to the binomial coefficient `n choose 2`, where `n` is
+ * the size of the underlying range. The reference type of this range is a tuple over the reference type of the
+ * underlying range. This range is const-iterable as long as the underlying range is iterable (const-ness of this range
+ * will not be propagated to the underlying range).
+ * In order to receive the end iterator in constant time an iterator pointing to the last element of the underlying
+ * range will be cached upon construction of this view. This construction takes linear time for underlying ranges that
+ * do not model std::ranges::BidirectionalRange.
+ *
+ * ### Iterator
+ *
+ * The returned iterator from begin does not model std::LegacyIterator, also known as `Cpp17Iterator` since it
```

Comment:
see above, I think it would be better to explicitly link to cppreferences as "std::LegacyIterator" is not really an entity.

Reply:
yes